hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/JetClassLoader.java

```diff
@@ -80,9 +80,37 @@ public JetClassLoader(
                     + " or start all members with it on classpath");
         }
         byte[] classBytes = uncheckCall(() -> IOUtil.toByteArray(classBytesStream));
+        definePackage(name);
         return defineClass(name, classBytes, 0, classBytes.length);
     }
 
+    /**
+     * Defines the package if it is not already defined for the given class
+     * name.
+     *
+     * @param className the class name
+     */
+    private void definePackage(String className) {
+        if (isEmpty(className)) {
+            return;
+        }
+        int lastDotIndex = className.lastIndexOf('.');
+        if (lastDotIndex == -1) {
+            return;
+        }
+        String packageName = className.substring(0, lastDotIndex);
+        if (getPackage(packageName) != null) {
+            return;
+        }
+        Package clPackage = JetClassLoader.class.getPackage();
+        try {
+            definePackage(packageName, clPackage.getSpecificationTitle(), clPackage.getSpecificationVersion(),
+                    clPackage.getSpecificationVendor(), clPackage.getImplementationTitle(),
+                    clPackage.getImplementationVersion(), clPackage.getImplementationVendor(), null);
```

Comment:
This is certainly not correct. I'm not sure but i think that the values need to be loaded from the manifest file, or rather we can just pass in nulls, as I saw in several examples on the web. We certainly should not pass jet's meta informations to user's packages.